setup/src/main/java/io/jenkins/jenkinsfile/runner/JenkinsfileRunnerLauncher.java

```diff
@@ -125,11 +127,7 @@ public int launch() throws Throwable {
         try {
             // so that test code has all the access to the system
             ACL.impersonate(ACL.SYSTEM);
-            ClassLoader cl = new ClassLoaderBuilder(jenkins.getPluginManager().uberClassLoader)
-                    .collectJars(new File(bootstrap.appRepo, "io/jenkins/jenkinsfile-runner/payload"))
-                    .make();
-
-            Class<?> c = cl.loadClass("io.jenkins.jenkinsfile.runner.Runner");
+            Class<?> c = bootstrap.hasClass(RUNNER_CLASS_NAME)? Class.forName(RUNNER_CLASS_NAME) : getRunnerClassFromJar();
```

Comment:
```suggestion
            Class<?> c = bootstrap.hasClass(RUNNER_CLASS_NAME) ? Class.forName(RUNNER_CLASS_NAME) : getRunnerClassFromJar();
```

You fetch your class twice here. IMHO it is fine for now, because I do not see a practical future for the old mode, except performance impact which needs to be measured. But this is a case where catching an exception would be better

Reply:
Hi @oleg-nenashev , do you want me to change it?